src/java.base/share/classes/java/lang/Math.java

```diff
@@ -2191,6 +2191,123 @@ public static double min(double a, double b) {
         return (a <= b) ? a : b;
     }
 
+    /**
+     * Clamps the value to fit between min and max. If the value is less
+     * than {@code min}, then {@code min} is returned. If the value is greater
+     * than {@code max}, then {@code max} is returned. Otherwise, the original
+     * value is returned.
+     * <p>
+     * While the original value of type long may not fit into the int type,
+     * the bounds have the int type, so the result always fits the int type.
+     * This allows to use method to safely cast long value to int with
+     * saturation.
+     *
+     * @param value value to clamp
+     * @param min minimal allowed value
+     * @param max maximal allowed value
+     * @return a clamped value that fits into {@code min..max} interval
+     * @throws IllegalArgumentException if {@code min > max}
+     *
+     * @since 21
+     */
+    public static int clamp(long value, int min, int max) {
+        if (min > max) {
+            throw new IllegalArgumentException(min + ">" + max);
```

Comment:
These should probably have some spacing or a better error message:
```suggestion
            throw new IllegalArgumentException(min + " > " + max);
```

Reply:
Updated, thanks